lib/ae_mdw/db/mutations/contract_call_mutation.ex

```diff
@@ -30,13 +43,47 @@ defmodule AeMdw.Db.ContractCallMutation do
 
   @spec mutate(t()) :: :ok
   def mutate(%__MODULE__{
+        contract_pk: contract_pk,
+        caller_pk: caller_pk,
         create_txi: create_txi,
         txi: txi,
         fun_arg_res: fun_arg_res,
         call_rec: call_rec
       }) do
     DBContract.call_write(create_txi, txi, fun_arg_res)
     DBContract.logs_write(create_txi, txi, call_rec)
+
+    with true <- Contract.is_aex9?(contract_pk),
+         {:ok, method_name, method_args} <-
+           Contract.extract_non_stateful_aex9_function(fun_arg_res) do
```

Comment:
is there a reason to use functions over events?

Reply:
The reason is to update aex9 presence based on balance after contract calls that may transfer funds and are out of the standard.

Reply:
good!